test/AspNetCore.HealthCheck.Tests/HealthCheckMiddlewareTests.cs

```diff
@@ -152,13 +159,47 @@ public class HealthCheckMiddlewareTests
 
             var defaultPolicy = new HealthCheckPolicy(new SettingsCollection());
 
-            var healthCheckMiddleware = new HealthCheckMiddleware(next, options.Object, loggerFactory, healthService.Object, defaultPolicy);
+            var authZService = new Mock<IAuthorizationService>();
+            var healthCheckMiddleware = new HealthCheckMiddleware(next, options.Object, loggerFactory, healthService.Object, defaultPolicy, authZService.Object);
             await healthCheckMiddleware.Invoke(contextMock.Object);
 
             healthService.Verify(s => s.CheckHealthAsync(It.IsAny<HealthCheckPolicy>()), Times.Once());
             Assert.Equal(StatusCodes.Status503ServiceUnavailable, contextMock.Object.Response.StatusCode);
         }
 
+        [Fact]
+        public async void Invoke_AuthZPolicyFailed_Returns404()
+        {
+            var contextMock = GetMockContext("/healthcheck");
+            RequestDelegate next = _ =>
+            {
+                return Task.FromResult<object>(null);
+            };
+
+            var options = new Mock<IOptions<HealthCheckOptions>>();
+            options.SetupGet(o => o.Value)
+                .Returns(new HealthCheckOptions()
+                {
+                    Path = "/healthcheck",
+                    AuthorizationPolicy = new AuthorizationPolicyBuilder().RequireClaim("invalid").Build()
+                });
+
+
+            var loggerFactory = new LoggerFactory();
+            var healthService = new Mock<IHealthCheckService>();
+            healthService.Setup(s => s.CheckHealthAsync(It.IsAny<HealthCheckPolicy>()))
+                .ReturnsAsync(new HealthResponse(new List<HealthCheckResult> { new HealthCheckResult { Status = HealthStatus.OK, } }));
+
+            var defaultPolicy = new HealthCheckPolicy(new SettingsCollection());
+
+            var authZService = new Mock<IAuthorizationService>();
+            var healthCheckMiddleware = new HealthCheckMiddleware(next, options.Object, loggerFactory, healthService.Object, defaultPolicy, authZService.Object);
+            await healthCheckMiddleware.Invoke(contextMock.Object);
+
+            healthService.Verify(s => s.CheckHealthAsync(It.IsAny<HealthCheckPolicy>()), Times.Never());
+            Assert.Equal(0, contextMock.Object.Response.StatusCode);
```

Comment:
404 is the default behavior done by the server. 
In fact , the middleware will not define a status code and delegate to the next middleware. 
Test name updated.